pom.xml

```diff
@@ -117,6 +117,7 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <maven.compiler.source>1.8</maven.compiler.source>
     <maven.compiler.target>1.8</maven.compiler.target>
+    <fabric8.mode>kubernetes</fabric8.mode>
```

Comment:
I'd prefer to add it directly to the f-m-p configuration with 

```xml
<configuration>
    <mode>kubernetes</mode>
    ...
</configuration>
```

It's close where it is used. Can still be overwritten from the outside. 

Plus I'm not 100% sure whether it breaks something different, but I _think_ that the CI build use source s2i (so not f-m-p at all), but not 100% sure. If it used binary s2i then setting this mode would prevent it. @iocanel is my assumption correct and is it save to nail the mode down to Kubernetes?